### PR TITLE
[WFLY-18258] AssumeTestGroupUtil should log exception if docker is unavailable and not assume false is ok

### PR DESCRIPTION
**Pull request by Dkafetzis**
https://issues.redhat.com/browse/WFLY-18258

### CODE REVIEW

File: testsuite/shared/src/main/java/org/jboss/as/test/shared/util/AssumeTestGroupUtil.java [hunk 1]
```diff
@@ -241,7 +244,15 @@ public static boolean isFullDistribution() {
      * @throws AssumptionViolatedException if a docker client cannot be initialized
      */
     public static void assumeDockerAvailable() {
-        assumeCondition("Docker is not available.", AssumeTestGroupUtil::isDockerAvailable);
+        try {
+            assumeCondition("Docker is not available.", AssumeTestGroupUtil::isDockerAvailable);
+        } catch (AssumptionViolatedException ex) {
+            if (System.getProperty("org.wildfly.test.require.docker").equals("true")){
```
Comment (bstansberry) [inline testsuite/shared/src/main/java/org/jboss/as/test/shared/util/AssumeTestGroupUtil.java:250]: I kicked off a discussion at https://wildfly.zulipchat.com/#narrow/stream/174184-wildfly-developers/topic/WFLY-18258.20--.20handling.20docker.20being.20unavailable.20in.20CI re whether we should do it like this or invert it. Absent any consensus to change this, this is fine.

**Review by bstansberry — CHANGES_REQUESTED** [review]
Thanks @Dkafetzis. I've suggested some changes below.

File: testsuite/shared/src/main/java/org/jboss/as/test/shared/util/AssumeTestGroupUtil.java [hunk 2]
```diff
@@ -254,6 +265,7 @@ public static boolean isDockerAvailable() {
             DockerClientFactory.instance().client();
             return true;
         } catch (Throwable ex) {
+            logger.error(String.format("Could not initialize docker, encountered exception Exception: %s", ex.getMessage()));
```
[the remark this answers is not in the document]
Comment (Dkafetzis) [reply testsuite/shared/src/main/java/org/jboss/as/test/shared/util/AssumeTestGroupUtil.java]: I agree, It has now been changed

File: testsuite/shared/src/main/java/org/jboss/as/test/shared/util/AssumeTestGroupUtil.java [hunk 3]
```diff
@@ -39,6 +41,7 @@ public class AssumeTestGroupUtil {
      * in-container where Shrinkwrap is not available.
      * </p>
      */
+    private static final Logger logger = Logger.getLogger(TestLogHandlerSetupTask.class);
```
[the remark this answers is not in the document]
Comment (Dkafetzis) [reply testsuite/shared/src/main/java/org/jboss/as/test/shared/util/AssumeTestGroupUtil.java]: Sorry, forgot to change it the first time

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @Dkafetzis

**Pull request merged**